exercises/practice/ledger/.meta/proof.ci.js

```diff
@@ -0,0 +1,67 @@
+class LedgerEntry {
+  constructor(date, description, change) {
+    this.date = new Date(date);
+    this.description = description;
+    this.change = change;
+  }
+}
+
+const headerRow = (locale) =>
+  locale === 'en-US'
+    ? `${`Date`.padEnd(10, ` `)} | ${`Description`.padEnd(
+        25,
+        ` `,
+      )} | ${`Change`.padEnd(13, ` `)}`
+    : `${`Datum`.padEnd(10, ` `)} | ${`Omschrijving`.padEnd(
+        25,
+        ` `,
+      )} | ${`Verandering`.padEnd(13, ` `)}`;
+
+export const createEntry = (date, description, change) =>
+  new LedgerEntry(date, description, change);
+
+export function formatEntries(currency, locale, entries) {
+  // Generate Header Row
+  let table = headerRow(locale);
+  // Set Formatting Options
+  let formattingOptions = {
+    style: 'currency',
+    currency: currency,
+    currencySign: locale === 'en-US' ? 'accounting' : 'standard',
+    currencyDisplay: locale === 'en-US' ? 'symbol' : 'narrowSymbol',
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  };
+  // Sort Entries
+  entries.sort(
+    (a, b) =>
+      a.date - b.date ||
+      a.change - b.change ||
+      a.description.localeCompare(b.description),
+  );
+  // Writing each entry to table
+  entries.forEach((entry) => {
+    table += '\n';
+    // Write entry date to table
+    table += `${entry.date.toLocaleDateString(locale, {
+      day: '2-digit',
+      month: '2-digit',
+      year: 'numeric',
+    })} | `;
+    // Write entry description to table
+    let truncatedDescription =
+      entry.description.length > 25
+        ? `${entry.description.substring(0, 22)}...`
+        : entry.description.padEnd(25, ' ');
+    table += `${truncatedDescription} | `;
+    // Write entry change to table
+    let changeStr = (entry.change / 100).toLocaleString(
+      locale,
+      formattingOptions,
+    );
+    let trailingSpace = changeStr.includes(')') ? '' : ' ';
+    table += `${changeStr}${trailingSpace}`.padStart(13, ' ');
+  });
+
+  return table;
```

Comment:
After my comments, I actually think what I would have done if I kept with the objects is something like:

```javascript
class LedgerEntry {
  constructor(date, description, change) {
    this.date = new Date(date);
    this.description = description;
    this.change = change;
  }
}

class FormattedLedgerEntry {
  constructor(entry, locale, dateFormat, currencyFormat) {
    this.entry = entry
    this.locale = locale;
    this.dateFormat = dateFormat
    this.currencyFormat = currencyFormat;
  }

  date(format = this.dateFormat) {
    return entry.date.toLocaleDateString(this.locale, format)
  }

  description(length = 25) {
    if (entry.description.length > length ) {
      return `${entry.description.substring(0, length - 3)}...`
    }

    return entry.description.padEnd(length, ' ');
  }
  
  change(offset = 13, format = this.currencyFormat) {
    const formatted = (entry.change / 100).toLocaleString(locale, format);
    
    const trailingSpace = formatted.includes(')') ? '' : ' ';
    `${formatted }${trailingSpace}`.padStart(offset, ' ');
  }

  toTableRow() {
    return [
      this.date(),
      this.description(),
      this.change()
    ].join(' | ')
  }
}

// Then:
const dateFormat = ...
const currencyFormat = ...

const rows = entries.map((entry) => {
  const formattedEntry = new FormattedLedgerEntry(entry, locale, dateFormat, currencyFormat))
  return entry.toTableRow()
})

return [
  headerRow(locale),
  ...rows
].join("\n")
```

Or something like that. 

Probably makes it much easier to maintain.

Reply:
I think I like your solution way more than mine 😁, so with your permission I'd borrow some of it, tho maybe when this exercise goes live.

I also tried to keep it functional, but I don't think it makes sense, particularly since I'm still using a `class`.

Reply:
Ya go for it! Use whatever you like. I was mostly just thinking of what I would've done